src/observer.rs

```diff
@@ -357,13 +356,8 @@ impl Observer {
             let register_resource = self.register_resources.get(register_resource_key).unwrap();
             let resource = self.resources.get(&register_resource.resource).unwrap();
 
-            let mut sequence_bin =
-                bincode::config().big_endian().serialize(&resource.sequence).unwrap();
-            let index = sequence_bin.iter().position(|&x| x > 0).unwrap();
-            sequence_bin.drain(0..index);
-
             message.set_token(register_resource.token.clone());
-            message.set_observe(sequence_bin);
+            message.set_observe(resource.sequence.to_be_bytes().to_vec());
```

Comment:
I think this is a bug, when I first did it I had malformed packets when inspecting them. Correct me if I'm wrong but `sequence` is `u32` which means 4 bytes when transformed into bytes whereas the standard allows 0-3 bytes. I think filtering only non-zero bytes is still necessary unfortunately

Reply:
You are right. Thanks. I fix it in #55